test/node.js

```diff
@@ -37,6 +44,75 @@ test('node pass on nodeOptions', async t => {
 	t.is(stdout, 'foo');
 });
 
+test.serial(
+	'node removes --inspect from nodeOptions when defined by parent process',
+	async t => {
+		process.execArgv = ['--inspect', '-e'];
+		const {stdout, stderr} = await execa.node('console.log("foo")', {
+			reject: false
+		});
+
+		t.is(stdout, 'foo');
+		t.is(stderr, '');
+	}
+);
```

Comment:
The tests (except the last one) are almost identical except for the value of `process.execArgv`.
Using [test macros](https://github.com/avajs/ava/blob/master/docs/01-writing-tests.md#reusing-test-logic-through-macros) might reduce code size here.